panic on parsing TX output datum

This block causes a panic when decoding the TX output datum.

https://cexplorer.io/block/26be71d8640226b327dc870e8b92d616046d336ece0759632b0509caa8c132ce

```
$ ./block-fetch -network mainnet -address relays-new.cardano-mainnet.iohk.io:3001 -ntn -hash 26be71d8640226b327dc870e8b92d616046d336ece0759632b0509caa8c132ce -s
lot 77420783 -all
era = Babbage, slot = 77420783, block_no = 8039509, id = 26be71d8640226b327dc870e8b92d616046d336ece0759632b0509caa8c132ce
...
panic: runtime error: hash of unhashable type []interface {}

goroutine 1 [running]:
type..hash.github.com/blinklabs-io/gouroboros/cbor.Value(0xc00006a600, 0xc0000de1c0?)
        <autogenerated>:1 +0x1e
reflect.mapassign(0x6a65c0, 0xc00006a680?, 0x0?, 0x6f5f61?)
        /usr/local/go/src/runtime/map.go:1353 +0x25
reflect.Value.SetMapIndex({0x6a65c0?, 0xc0000140c8?, 0xc00006a680?}, {0x6c3dc0, 0xc00006a600, 0x199}, {0x6c3dc0, 0xc00006a680, 0x199})
        /usr/local/go/src/reflect/value.go:2325 +0x391
...
```

--- a/cbor/value.go
+++ b/cbor/value.go
@@ -14,6 +14,10 @@
 
 package cbor
 
+import (
+	"fmt"
+)
+
 // Helpful wrapper for parsing arbitrary CBOR data which may contain types that
 // cannot be easily represented in Go (such as maps with bytestring keys)
 type Value struct {
@@ -22,12 +26,20 @@ type Value struct {
 	cborData string
 }
 
-func (v *Value) UnmarshalCBOR(data []byte) error {
+func (v *Value) UnmarshalCBOR(data []byte) (err error) {
 	// Save the original CBOR
 	v.cborData = string(data[:])
 	cborType := data[0] & CBOR_TYPE_MASK
 	switch cborType {
 	case CBOR_TYPE_MAP:
+		// There are certain types that cannot be used as map keys in Go but are valid in CBOR. Trying to
+		// parse CBOR containing a map with keys of one of those types will cause a panic. We setup this
+		// deferred function to recover from a possible panic and return an error
+		defer func() {
+			if r := recover(); r != nil {
+				err = fmt.Errorf("decode failure, probably due to type unsupported by Go: %v", r)
+			}
+		}()
 		tmpValue := map[Value]Value{}
 		if _, err := Decode(data, &tmpValue); err != nil {
 			return err

